tests/integration/test_records.py

```diff
@@ -306,6 +307,26 @@ def test_merged_records_stay_merged(api_client, merged_records):
     assert api_client.get('/literature/222').status_code == 301
 
 
+def test_merge_record_with_non_existing_pid(api_client, merged_records):
+    def get_pid_entry(recid):
+        return PersistentIdentifier.query.filter_by(pid_value=str(recid)).one_or_none()
+
+    merged_record = get_db_record('lit', 111)
+
+    assert get_recid_from_ref(merged_record['deleted_records'][0]) == 222
+    # remove it so it doesn't exists anymore
+    db.engine.execute("delete from pidstore_pid where pid_value='222'")
```

Comment:
Maybe you should use here the 'usual' way to delete a record, as it might do something more than just delete a row in a table.

Reply:
The problem is that ``delete`` and ``_delete`` functions delete the record but not the PID, which is exactly what I need.

Reply:
Then you might want to use the pid model then? (to delete the pid)

Reply:
Same, ``delete`` marks the PID as deleted (if it's already registered as in this case).
Any suggestion?

https://invenio-pidstore.readthedocs.io/en/latest/api.html#module-invenio_pidstore.models